### ISSUE TITLE
Packaging discussion about xapi.conf

### ISSUE THREAD

**Issue by stormi** [xapi-project/xen-api#4181]
I'm not sure this is a question for the XAPI project itself because the RPM spec file isn't in the repository, but I think the involved people are the same and I've got little results with questions on bugs.xenserver.org.

In Citrix Hypervisor's `xapi` SRPM, `xapi.conf` is defined as `%config(noreplace)`, which means that updates of the RPM will not override changes made by the user to that file. It makes it difficult to update the default values it contains.

Is there a specific reason for the current definition? It looks like, to me, that users are not really supposed to touch that file, especially when they have the ability to drop a file into `/etc/xapi.conf.d/` instead.

Wouldn't it be better to make it simply `%config`, which means that user modifications will be saved into a `.rpmsave` file if needed, while guaranteeing that any change you bring to the defaults in xapi.conf are applied to users who upgrade using an update ISO? I'd also add a warning on top of `xapi.conf` to explain that this file is not meant to be modified (actually I would even move it out of `/etc` if possible) and that user settings belong to `/etc/xapi.conf.d/`.

**Comment by edwintorok** [comment]
Usually new values have defaults in xapi, so if the entry is missing from xapi.conf it'll use the value from the binary.

If you want to ship different defaults from the one in the binary you can put them into `/etc/xapi.conf.d` as you said, with a low enough sort order to allow the user to still override it. That would also work for updating defaults (although it'll override any user customizations).

To avoid problems with hotfixes/updates overwriting a user's xapi.conf I think it'd be useful to force them to use the `/etc/<service>.conf.d/` files instead. Perhaps a new version of xapi could ship a version of xapi.conf all commented out, and would refuse to start if anything in the file is uncommented (requiring users to use .conf.d instead?).
Distribution specific customizations could also go into a `/etc/xapi.conf.d/00-distribution.conf`. 
Similarly for xenopsd and other daemons, this is handled in a common place in xapi-idl.

**Comment by stormi** [comment]
I agree with most of this, but it feels a bit awkward to have a configuration file that is not meant to be modified (`xapi.conf`). To me, either it can be modified or it doesn't need to exist. I understand that it could be useful as documentation, but then it doesn't belong to `/etc` and should not be read by XAPI at all. As you said, XAPI already knows about the defaults.

Ideally, distribution specific settings should also not belong to `/etc`. I'd rather have a `xapi-defaults.conf` file somewhere in `/usr`.
